lib/req/request.ex

```diff
@@ -408,11 +408,15 @@ defmodule Req.Request do
         |> Keyword.update(:options, %{}, &Map.new/1)
         |> Keyword.update(
           :registered_options,
-          MapSet.new([:redact_auth]),
-          &MapSet.put(&1, :redact_auth)
-        )
+          MapSet.new([:redact_headers, :redact_options]),
+          fn registered_options -> 
```

Comment:
Perhaps, this can be passed in as a reference with `&`. I think it would make the code a bit cleaner.